### PR TITLE
[WIP] Add initial Kirigami GUI

### PR DESCRIPTION
**Pull request by jbruechert**
* Banner image still to be chosen

Not yet checked for correct coding style

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -59,6 +59,14 @@ int main(int argc, char *argv[])
 	// QML-GUI
 	//
 
+	if (strlen(qgetenv("QT_QUICK_CONTROLS_STYLE")) == 0)
```
Comment (lnjX) [inline src/main.cpp]: Please use `qgetenv("QT_QUICK_CONTROLS_STYLE").isEmpty()`; AFAIK `strlen()` is a C function and using Qt's own function on a QByteArray is safer.

Alternatively, you could also do `qgetenv("QT_QUICK_CONTROLS_STYLE").trimmed().isEmpty()`, that would first remove all characters like spaces, tabs (at the beginning) and newlines (beginning/end) and then check the size.

File: src/main.cpp [hunk 2]
```diff
@@ -59,6 +59,14 @@ int main(int argc, char *argv[])
...
+	{
+		qDebug() << "QT_QUICK_CONTROLS_STYLE not set, setting to Material";
+		qputenv("QT_QUICK_CONTROLS_STYLE", "Material");
+	}
+	qputenv("QT_QUICK_CONTROLS_1_STYLE", "Flat");
```
Comment (lnjX) [inline src/main.cpp]: Shouldn't we set it only if empty, too?

File: src/qml/VertretungsplanPage.qml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			title: "Gruppe"
+			width: 200
+		}
+
+		anchors.fill: parent
+
+		model: ListModel {
+			ListElement {
+				lesson: "1"
+				group: "10.21"
```
Comment (lnjX) [inline src/qml/VertretungsplanPage.qml:52]: Oh nooo nicht die 21 ...

**Comment by lnjX** [comment]
Of course, it's nice to have a banner, but I don't think it's necessary to really add one. But you could take a GHO logo on a background (maybe a simple colour gradient) ...

File: src/main.cpp (hunk 2 above)
Comment (jbruechert) [reply src/main.cpp]: I only use QQC 1 for the table which I would like to have in this design

File: src/main.cpp (hunk 1 above)
Comment (jbruechert) [reply src/main.cpp]: nice, I still have to learn to just use Qt stuff for everything :)

File: src/qml/VertretungsplanPage.qml (hunk 3 above)
Comment (jbruechert) [reply src/qml/VertretungsplanPage.qml:52]: It's a fake model anyway :)

File: src/main.cpp [hunk 4]
```diff
@@ -59,6 +59,14 @@ int main(int argc, char *argv[])
 	// QML-GUI
 	//
 
+	if (qgetenv("QT_QUICK_CONTROLS_STYLE").isEmpty())
+	{
```
Comment (lnjX) [inline src/main.cpp]: ah and don't do this; only put the `{` into a new line for functions; Of course this differs from project to project, but I'd like to do it like in the kernel here.

File: src/qml/VertretungsplanPage.qml (hunk 3 above)
Comment (lnjX) [reply src/qml/VertretungsplanPage.qml:52]: :rofl:

**Pull request merged**